pkg/cmd/repo/clone/clone.go

```diff
@@ -131,6 +142,11 @@ func cloneRun(opts *CloneOptions) error {
 	}
 	canonicalCloneURL := ghrepo.FormatRemoteURL(canonicalRepo, protocol)
 
+	// If RepoHasWiki is true and repo opts has wiki then create a new clone URL
+	if canonicalRepo.RepoHasWiki() && wikipref {
```

Comment:
This condition is wrong. If `wikipref` is true, the user has indicated that they want to clone a wiki, so we must always attempt to clone the wiki. However, if the current repository does not have a wiki, we need to throw an error. Something like:
```go
if wikipref {
  if !canonicalRepo.HasWikiEnabled {
    return fmt.Errorf("the '%s' repository does not have a wiki", ghrepo.FullName(canonicalRepo))
  }
  // ...
}
```